apm-agent-plugins/apm-jmx-plugin/src/main/java/co/elastic/apm/agent/jmx/JmxMetricTracker.java

```diff
@@ -83,50 +96,131 @@ public void onChange(ConfigurationOption<?> configurationOption, List<JmxMetric>
-                    final CompositeData compositeValue = (CompositeData) value;
-                    for (final String key : compositeValue.getCompositeType().keySet()) {
-                        if (compositeValue.get(key) instanceof Number) {
-                            registrations.add(new JmxMetricRegistration(JMX_PREFIX + attribute.getMetricName() + "." + key, Labels.Mutable.of(objectName.getKeyPropertyList()), attribute.getJmxAttributeName(), key, objectName));
-                        } else {
-                            logger.warn("Can't create metric '{}' because composite value '{}' is not a number: '{}'", jmxMetric, key, value);
+                final Object value;
+                try {
+                    value = server.getAttribute(objectName, attribute.getJmxAttributeName());
+                    if (value instanceof Number) {
+                        logger.debug("Found number attribute {}={}", attribute.getJmxAttributeName(), value);
+                        registrations.add(new JmxMetricRegistration(JMX_PREFIX + attribute.getMetricName(),
+                            Labels.Mutable.of(objectName.getKeyPropertyList()),
+                            attribute.getJmxAttributeName(),
+                            null,
+                            objectName));
+                    } else if (value instanceof CompositeData) {
+                        final CompositeData compositeValue = (CompositeData) value;
+                        for (final String key : compositeValue.getCompositeType().keySet()) {
+                            if (compositeValue.get(key) instanceof Number) {
+                                logger.debug("Found composite number attribute {}.{}={}", attribute.getJmxAttributeName(), key, value);
+                                registrations.add(new JmxMetricRegistration(JMX_PREFIX + attribute.getMetricName() + "." + key,
+                                    Labels.Mutable.of(objectName.getKeyPropertyList()),
+                                    attribute.getJmxAttributeName(),
+                                    key,
+                                    objectName));
+                            } else {
+                                logger.warn("Can't create metric '{}' because composite value '{}' is not a number: '{}'", jmxMetric, key, value);
+                            }
                         }
+                    } else {
+                        logger.warn("Can't create metric '{}' because attribute '{}' is not a number: '{}'", jmxMetric, attribute.getJmxAttributeName(), value);
                     }
-                } else {
-                    logger.warn("Can't create metric '{}' because attribute '{}' is not a number: '{}'", jmxMetric, attribute.getJmxAttributeName(), value);
+                } catch (AttributeNotFoundException e) {
+                    logger.warn("Can't create metric '{}' because attribute '{}' could not be found", jmxMetric, attribute.getJmxAttributeName());
```

Comment:
why not return 'false' here when there is an error ? Would that make retry forever in case of error ?

Reply:
It means that the MBean itself was found but that the attribute was not. This is likely a misspelling of the attribute name.

> Would that make retry forever in case of error ?

Yes, that's what I think